scipy/spatial/transform/tests/test_rigid_transform.py

```diff
@@ -1002,28 +1373,35 @@ def test_empty_transform_concatenation():
     assert len(RigidTransform.concatenate([tf_many, tf_empty, tf_single])) == 3
 
 
-def test_empty_transform_inv_and_pow():
-    tf = RigidTransform.identity(0)
+def test_empty_transform_inv_and_pow(xp):
+    tf = RigidTransform.from_matrix(xp.empty((0, 4, 4)))
     assert len(tf.inv()) == 0
-    assert len(tf ** 0) == 0
-    assert len(tf ** 1) == 0
-    assert len(tf ** -1) == 0
-    assert len(tf ** 0.5) == 0
+    assert len(tf**0) == 0
+    assert len(tf**1) == 0
+    assert len(tf**-1) == 0
+    assert len(tf**0.5) == 0
 
 
-def test_empty_transform_indexing():
+def test_empty_transform_indexing(xp):
     tf_many = RigidTransform.identity(3)
-    tf_zero = tf_many[[]]
+    tf_many = RigidTransform.from_matrix(xp.asarray(tf_many.as_matrix()))
+    tf_zero = tf_many[xp.asarray([], dtype=xp.int64)]
     assert len(tf_zero) == 0
 
-    assert len(tf_zero[[]]) == 0
-    assert len(tf_zero[:5]) == 0  # Slices can go out of bounds.
+    assert len(tf_zero[xp.asarray([], dtype=xp.int64)]) == 0
+    # Array API does not specify out-of-bounds indexing. Only check for numpy.
+    if is_numpy(xp):
+        assert len(tf_zero[:5]) == 0  # Slices can go out of bounds.
 
     with pytest.raises(IndexError):
         tf_zero[0]
 
-    with pytest.raises(IndexError):
-        tf_zero[[0, 2]]
+    if is_jax(xp):  # DECISION: Should we manually check and raise for jax here?
+        with pytest.raises(TypeError, match="Slice size at index 0 in gather op"):
+            tf_zero[xp.asarray([0, 2])]
+    else:
+        with pytest.raises(IndexError):
+            tf_zero[xp.asarray([0, 2])]
```

Comment:
I assume this is because of https://docs.jax.dev/en/latest/notebooks/Common_Gotchas_in_JAX.html#out-of-bounds-indexing?
Does the out of bounds index derive from a Python scalar or from the values of the input array (e.g. via `xp.take`)? In the former case, couldn't your own Python code check it?